src/shared/components/oncoprint/tabularDownload.ts

```diff
@@ -78,75 +79,66 @@ export function getTabularDownloadData(
-            oncoprintData.CNA[currentGeneName][id] = '';
-            oncoprintData.MUTATIONS[currentGeneName][id] = '';
-            oncoprintData.MRNA[currentGeneName][id] = '';
-            oncoprintData.PROTEIN[currentGeneName][id] = '';
-            oncoprintData.STRUCTURAL_VARIANT[currentGeneName][id] = '';
-            if (geneticTrackDatum.disp_cna !== undefined) {
-                oncoprintData.CNA[currentGeneName][id] = cnaMap[
-                    geneticTrackDatum.disp_cna
-                ]
-                    ? cnaMap[geneticTrackDatum.disp_cna]
-                    : geneticTrackDatum.disp_cna;
-            }
-            if (geneticTrackDatum.disp_structuralVariant !== undefined) {
-                oncoprintData.STRUCTURAL_VARIANT[currentGeneName][
-                    id
-                ] = structuralVariantMap[
-                    geneticTrackDatum.disp_structuralVariant + ''
-                ]
-                    ? structuralVariantMap[
-                          geneticTrackDatum.disp_structuralVariant + ''
-                      ]
-                    : geneticTrackDatum.disp_structuralVariant;
-            }
-            if (geneticTrackDatum.disp_mrna !== undefined) {
-                oncoprintData.MRNA[currentGeneName][id] = mrnaMap[
-                    geneticTrackDatum.disp_mrna
-                ]
-                    ? mrnaMap[geneticTrackDatum.disp_mrna]
-                    : geneticTrackDatum.disp_mrna;
-            }
-            if (geneticTrackDatum.disp_prot !== undefined) {
-                oncoprintData.PROTEIN[currentGeneName][id] = proteinMap[
-                    geneticTrackDatum.disp_prot
-                ]
-                    ? proteinMap[geneticTrackDatum.disp_prot]
-                    : geneticTrackDatum.disp_prot;
+
+            // Only initialize each alteration for sample/patient if it doesn't exist
+            // It may already exist if there are multiple OncoPrint tracks of the same gene
+            for (const alteration in alterationsMap) {
+                if (
+                    oncoprintData[alteration][currentGeneName][id] === undefined
+                ) {
+                    oncoprintData[alteration][currentGeneName][id] = '';
+                }
             }
-            if (geneticTrackDatum.disp_mut !== undefined) {
-                oncoprintData.MUTATIONS[currentGeneName][id] = mutationMap[
-                    geneticTrackDatum.disp_mut
-                ]
-                    ? mutationMap[geneticTrackDatum.disp_mut]
-                    : geneticTrackDatum.disp_mut;
+
+            // Check if the sample/patient is initialized, and combine both results in case they are different
+            for (const alteration in alterationsMap) {
+                var disp = alterationsMap[alteration] as keyof typeof string;
+                if (geneticTrackDatum[disp] !== undefined) {
+                    var current_value = cnaMap[geneticTrackDatum[disp]]
```

Comment:
@oplantalech should this really be cnaMap for all alteration type?

Reply:
Thanks, fixed